backend/src/main/java/com/google/blackswan/mock/AdvanceAlertGenerator.java

```diff
@@ -0,0 +1,53 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.blackswan.mock;
+
+import com.google.models.*;
+import java.util.List;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ListMultimap;
+import com.google.common.collect.LinkedListMultimap;
+
+/** Generate list of alerts of topics requested by grouping anomalies by their month. */
+public class AdvanceAlertGenerator implements AlertGenerator {
```

Comment:
maybe instead of `Advance` you can rename it to be something mroe specific (ie more related to this monthly grouping)

Reply:
I renamed it to `MultiInputAlertGenerator` since the main difference with this implementation of `AlertGenerator` is that it takes in multiple `DataInfo topic`s as inputs and doesn't just generate alert for one specific topic. 

All of the alert generators currently group anomalies by month, so i think the `MultiInput...` naming might be better. Thanks for the suggestion!